src/js/components/Box/StyledBox.js

```diff
@@ -222,22 +230,33 @@ const gapStyle = (directionProp, gap, responsive, wrap, theme) => {
       );
       if (responsiveMetric) {
         styles.push(
-          breakpointStyle(breakpoint, `column-gap: ${responsiveMetric};`),
+          breakpointStyle(
+            breakpoint,
+            `column-gap: ${responsiveMetric};`,
+            responsive,
+          ),
         );
       }
     }
   } else if (directionProp === 'column' || directionProp === 'column-reverse') {
     styles.push(`row-gap: ${metric};`);
     if (responsiveMetric) {
-      styles.push(breakpointStyle(breakpoint, `row-gap: ${responsiveMetric};`));
+      styles.push(
+        breakpointStyle(breakpoint, `row-gap: ${responsiveMetric};`),
+        responsive,
```

Comment:
Should this be 
```suggestion
        breakpointStyle(breakpoint, `row-gap: ${responsiveMetric};`, responsive),
```

Reply:
Oooo great catch. A couple of spots just above too for gap. And there's also a pre-existing bug here we should fix, maybe as a separate PR? If gap is an object like `gap={{ column: "medium", row: "xsmall" }}` then line `responsiveMetric` at line 200 is always `undefined`.

Reply:
good to know, I feel like since it is a pre-existing issue it could be a separate PR